agent/xds/testdata/listeners/connect-proxy-with-chain-external-sni.golden

```diff
@@ -16,7 +16,7 @@
             {
               "name": "envoy.tcp_proxy",
               "config": {
-                  "cluster": "",
+                  "cluster": "db.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul",
```

Comment:
These tests do cover the actual behaviour here but given that we didn't spot that the golden config was broken in these cases I wonder if it would be a good idea to add an integration test that exercises this path explicitly that would have cause this?

If it is more than a few minutes to get something working though maybe we could land this as it is and make an issue to come back to it as I think you have more important work and I'd like this fix in the next release for sure!

Reply:
Done.